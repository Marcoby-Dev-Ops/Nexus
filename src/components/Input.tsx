import React from 'react';
import PropTypes from 'prop-types';

/**
 * @interface InputProps
 * @description Props for the Input component.
 */
export interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  error?: string;
  icon?: React.ReactNode;
}

/**
 * @name Input
 * @description A styled input field with error and icon support.
 * @param {InputProps} props - The props for the component.
 * @returns {JSX.Element} The rendered Input component.
 */
export const Input: React.FC<InputProps> = React.memo(({ error, icon, className = '', ...props }) => (
  <div className="flex flex-col gap-1">
    <div className={`relative flex items-center ${error ? 'border-destructive' : 'border-input'} border rounded-md px-3 py-2 bg-background ${className}`}>
      {icon && <span className="mr-2 text-muted-foreground">{icon}</span>}
      <input
        className="flex-1 bg-transparent outline-none"
        aria-invalid={!!error}
        {...props}
      />
    </div>
    {error && <span className="text-xs text-destructive mt-1">{error}</span>}
  </div>
));

Input.displayName = 'Input';

Input.propTypes = {
  error: PropTypes.string,
  icon: PropTypes.node,
  className: PropTypes.string,
};

export default Input; 