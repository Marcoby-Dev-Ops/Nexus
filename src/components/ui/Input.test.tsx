/**
 * @file Input.test.tsx
 * @description Unit and snapshot tests for the Input component.
 */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Input from './Input';

describe('Input', () => {
  it('renders with placeholder', () => {
    render(<Input placeholder="Type here" onChange={jest.fn()} />);
    expect(screen.getByPlaceholderText('Type here')).toBeInTheDocument();
  });

  it('calls onChange when value changes', () => {
    const handleChange = jest.fn();
    render(<Input placeholder="Type here" onChange={handleChange} />);
    fireEvent.change(screen.getByPlaceholderText('Type here'), { target: { value: 'Hello' } });
    expect(handleChange).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Input placeholder="Type here" onChange={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });
}); 