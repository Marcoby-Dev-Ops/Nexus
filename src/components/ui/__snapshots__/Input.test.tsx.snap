// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-1"
  >
    <div
      class="relative flex items-center border-input border rounded-md px-4 py-4 bg-background "
    >
      <input
        aria-invalid="false"
        class="flex-1 bg-transparent outline-none"
        placeholder="Type here"
      />
    </div>
  </div>
</DocumentFragment>
`;
